File: Frontend/src/services/authService.js
import api from "./api";

// Admin login
export const adminLogin = async (loginData) => {
  try {
    const response = await api.post("/auth/adminLogin", loginData);
    return response.data;
  } catch (error) {
    console.error("Error during admin login:", error);
    throw error;
  }
};

// Admin register
export const adminRegister = async (formData) => {
  try {
    const response = await api.post("/auth/adminRegister", formData, {
      headers: { "Content-Type": "application/json" },
    });
    console.log("Server response:", response);
    return response.data;
  } catch (error) {
    console.error("Error during admin registration:", error);
    throw error;
  }
};


// Send email OTP
export const sendEmailOtp = async (idf) => {
  try {
    const response = await api.post("/auth/sendEmailOtp", { idf });
    return response.data;
  } catch (error) {
    console.error("Error sending email OTP:", error);
    throw error;
  }
};

// Verify OTP
export const verifyOtp = async (idf, otp) => {
  try {
    const response = await api.post("/auth/verifyOtp", { idf, otp });
    return response.data;
  } catch (error) {
    console.error("Error verifying OTP:", error);
    throw error;
  }
};

File: Backend/src/main/java/com/EmployeeManagement/Employee/Management/Service/AuthService.java
package com.EmployeeManagement.Employee.Management.Service;

import com.EmployeeManagement.Employee.Management.Entity.Admins;
import com.EmployeeManagement.Employee.Management.RequestEntity.LoginReq;
import com.EmployeeManagement.Employee.Management.response.Response;

public interface AuthService {

    Response adminRegister(Admins admin);

    String sendEmailOtp(String email);

    String verifyOtp(String idf, String otp);

    Response adminLogin(LoginReq loginReq);

    void sendmail(String toEmail, String subject, String data);
}
